perf/change_point_detector.go

```diff
@@ -18,7 +18,7 @@ import (
 )
 
 type ChangeDetector interface {
-	DetectChanges(context.Context, []float64) ([]model.ChangePoint, error)
+	DetectChanges(context.Context, []float64, string) ([]model.ChangePoint, error)
```

Comment:
can't there be documentation here. I wouldn't change the interface just to initialize the data.

Reply:
I can document it; seems simpler to specify the measurement you are detecting a change for rather than fixing up the result elsewhere in the codebase (which can be forgotten).

Reply:
right but, this forces us to leak the data limitation into the implementation of this interface. 

In retrospect, it's probably a problem that we return a type from the model package at all.